.github/workflows/eks-deploy.yaml

```
@@ -0,0 +1,109 @@
+name: eks-deploy
+
+permissions:
+  id-token: write
+  contents: read
+  deployments: write
+
+on:
+  workflow_call:
+    inputs:
+      app-name:
+        description: 'The name of the application.'
+        required: true
+        type: string
+      environment:
+        description: 'The environment to deploy to.'
+        required: true
+        type: string
+      cluster-name:
+        description: 'The name of the cluster to deploy to.'
+        required: false
+        type: string
+        default: web
+      namespace:
+        description: 'The namespace to deploy to.'
+        type: string
+        default: hubs
+      ytt-template:
+        description: 'Kubernetes configuration template to use'
+        required: false
+        type: string
+        default: service-v1
+      image-ref:
+        description: 'The image reference to deploy.'
+        required: true
+        type: string
+      deploy:
+        description: 'Actually deploy, instead of only showing diff'
+        required: false
+        type: boolean
+        default: false
+      devops-ref:
+        description: 'The ref of the devops repository to use.'
+        required: false
+        type: string
+        default: master
+
+    secrets:
+      PRIVATE_REPO_TOKEN:
+        required: true
+
+jobs:
+  eks-deploy:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+    steps:
+      - name: Setup carvel
+        uses: carvel-dev/setup-action@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          only: ytt, kapp
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          path: service
+      - name: Checkout devops
+        uses: actions/checkout@v3
+        with:
+          repository: 3DHubs/devops
+          ref: ${{ inputs.devops-ref }}
+          token: ${{ secrets.PRIVATE_REPO_TOKEN }}
+          path: devops
+      - name: Cluster auth
+        id: eks-auth
+        uses: 3DHubs/workflows/.github/actions/eks-auth@remove-eks-auth-event-check
```

```suggestion
        uses: 3DHubs/workflows/.github/actions/eks-auth@main
```
We should merge @vixus0's changed to the eks-auth first